utils/auth-utils.js

```diff
@@ -22,22 +38,33 @@ async function comparePassword(password, hash) {
   throw new AuthenticationError(400, 'Username or Password is incorrect!');
 };
 
+/**
+ * validates password against set rules
+ * @param {String} password plaintext password
+ */
 function validatePassword(password) {
   const regexp = /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9]).{8,}$/;
 
   const valid = regexp.test(password);
 
   if (!valid) {
-    throw new ValidationError(400, 'Please ensure your password satisfies the following', [
-      'At least 1 Uppercase Letter',
-      'At least 1 Lowercase Letter',
-      'At least 1 Number',
-    ]);
+    throw new ValidationError(
+        400,
+        'Please ensure your password satisfies the following',
+        [
+          'At least 1 Uppercase Letter',
+          'At least 1 Lowercase Letter',
+          'At least 1 Number',
+        ]
+    );
   }
 }
-
+/**
+ * validates email against regexp
+ * @param {String} email email address
+ */
 function validateEmail(email) {
-  const regexp = /(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])/;
+  const regexp = /\b[\w.%-]+@[\w.-]+.[a-z]{2,}\b/;
```

Comment:
WOW, now that's an improvement! However, I did find a slight bug while testing this out in a regexr tester website (https://regexr.com). It seems to prematurely match the email without a domain. For example, if I type in john.snow@somedomain => this results in a match, whereas the other (albeit quite a bit a more uglier), results in a no match. https://imgur.com/a/yEujx9P

Reply:
Probably not an issue, but just in case.

Reply:
Thanks for catching that man! That's a big bug haha! I just left out a backslash :stuck_out_tongue_closed_eyes:

Reply:
ahhh, sweet! Yeah thought was just a simple typo!